runners/flink/1.10/job-server/build.gradle

```diff
@@ -29,3 +29,7 @@ project.ext {
 
 // Load the main build script which contains all build logic.
 apply from: "$basePath/flink_job_server.gradle"
+
+dependencies {
+  permitUnusedDeclared project(":runners:flink:1.10")
```

Comment:
I think a better solution is to make `:runners:flink:1.x` a runtimeOnly instead of dependency instead of a compile dependency in flink_job_server.gradle, but I will do that in a separate PR since more changes might be required. BEAM-11664

Reply:
done.